src/pymor/algorithms/symplectic.py

```diff
@@ -0,0 +1,475 @@
+# This file is part of the pyMOR project (https://www.pymor.org).
+# Copyright pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (https://opensource.org/licenses/BSD-2-Clause)
+
+import numpy as np
+from scipy.linalg import schur
+
+from pymor.algorithms.pod import pod
+from pymor.core.base import BasicObject
+from pymor.core.defaults import defaults
+from pymor.core.exceptions import AccuracyError
+from pymor.core.logger import getLogger
+from pymor.operators.symplectic import CanonicalSymplecticFormOperator
+from pymor.vectorarrays.block import BlockVectorSpace
+from pymor.vectorarrays.constructions import cat_arrays
+from pymor.vectorarrays.interface import VectorArray
+
+
+class SymplecticBasis(BasicObject):
```

Comment:
Does this class behave like other `VectorArrays`? If so, it might be better to move it to `src/pymor/vectorarrays/symplectic.py` and rename it to something like `SymplecticVectorArray`.

Reply:
Indeed, a SymplecticBasis is a container for pairs of basis vectors which are currently stored in two separate VectorArrays.  In that sense it is a VectorArray with a special structure and which allows for further functionalities. For technical reasons it is helpful to keep the pairs in separate blocks. Thus, the vectors cannot be saved to a single VectorArray, as `append` would require to append vectors "in the middle" of the VectorArray. If the blocks are stored in two separate lists, the problem is that in some cases they should be treated as a single VectorArray (e.g. on `lincomb`, `axpy` etc) and all these operations become ugly duplicate code since the operation has to be split up for the two VectorArrays. Thus, the current version implements it as a new object, which, when the functionality as "one VectorArray" uses the "to_array" function.

Reply:
Ok, thanks. Then, does it make sense to implement `SymplecticBasis` as a special case of a `BlockVectorArray`, where there are exactly two blocks coming from the same `VectorSpace`?
It might make it easier to see some equations. Could you add references to the algorithms you are implementing?

Reply:
`BlockVectorArray` would not be suitable. Look at the `lincomb` method, for instance.

Reply:
I see, thanks, that's interesting. Then it works like a pair of `VectorArrays`, with some properties. I guess if we find another situation where a collection of `VectorArrays` needs to be handled as a single object, we might think about more general interfaces, but that doesn't seem necessary at this point.